pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -163,14 +163,71 @@ in
     nvimRequireCheck = "autosave";
   };
 
-  avante-nvim = (callPackage ./avante-nvim { }).overrideAttrs {
-    dependencies = with self; [
-      dressing-nvim
-      nui-nvim
-      nvim-treesitter
-      plenary-nvim
-    ];
-  };
+  avante-nvim =
```

Comment:
quick question, why do we derive away from other plugins?

afaict every other plugin is using overrideAttrs? Wouldn't this make it easier to maintain with what we have like before?

Reply:
This plugin isn't part of the generated.nix like the others. I'm not familiar with the history of this plugin and why it's doing it this way... but might be nice to just setup a derivation for `avante-nvim-lib` and then move `avante.nvim` to behave similar to the others and use overrides.nix to handle just the postInstall and dependencies

Reply:
If the lib is built from the same source as the vimPlugin then you could just have the main plugin source get updated in generated.nix and the overrideAttrs in here could do the `avante-nvim-lib` build for the postInstall.

Reply:
quick context: we have a few bindings in rust, therefore we probably need a `avante-nvim-lib` derivation (similar to what you suggest here)

otherwise I think your suggestion makes sense

Reply:
I am the one who packaged this plugin if I remember well.
First, the plugin has semantic releases, which encourages for using those instead of the latest git commit.
Also, as it needed some more work in `overrides.nix`, it was logical to have it completely packaged here.

There are examples of plugins which build some rust binaries while having their source updated using the script (part of `generated.nix`). The issue is that if someone updates the `vimPlugins` package set without paying attention, the `carhoHash` can become outdated and thus the plugin will not build anymore.

Reply:
> First, the plugin has semantic releases, which encourages for using those instead of the latest git commit.

Well, I made a mistake and thought about another plugin. This plugin has no semantic releases.
It would then make more sense to have its source being updated as part of the `generated.nix` process.

@ttrei, could you go ahead and modify `vim-plugin-names`, `generated.nix` and `overrides.nix` accordingly ?

Reply:
(Also, I am not the one who has packaged it in nixpkgs, but I added the nixvim module).

Reply:
> @ttrei, could you go ahead and modify `vim-plugin-names`, `generated.nix` and `overrides.nix` accordingly ?

Here is a proposition: https://github.com/NixOS/nixpkgs/commit/4a7dfbd96f53a57c3f1fc5875911a9fc8c562125
If everything looks good to you, just pick the changes and update your branch.

Reply:
@GaetanLepage, you removed the meta.maintainers and other attributes, is this intentional?

Reply:
Indeed, this is a mistake.
Fixed in https://github.com/GaetanLepage/nixpkgs/commit/53a3522af3e0b4fb43c5a57e9ff41a96047a75f0.

Reply:
thanks! included your changes in the pr

do i understand correctly that now the plugin version will get updated during mass plugin updates?

Reply:
> do i understand correctly that now the plugin version will get updated during mass plugin updates?

Yes. When running the script though, `cargoHash` will have to be updated manually.
This would be surely caught when running `nixpkgs-review` so I am not too worried.

Reply:
> cargoHash will have to be updated manually.

This would be really annoying, maybe return to the previous method but just add an update script?